apple/swift-docc
# Fix a bug where the diagnostics file dropped all conversion problems

Bug/issue #, if applicable: rdar://115521475

## Summary

This fixes a bug where the DiagnosticsFileWriter dropped all the problems from the documentation conversion because `flush()` is called twice and the file writer assumed that it would only be called once and removed all the gathered problem after writing them to a file. This resulted in the next `flush()` call to write an empty diagnostics file.

## Dependencies

None

## Testing

- Build documentation for some project with documentation warnings (for example unresolved symbol links) and pass \
   `--diagnostics-file /path/to/something.json`

- Inspect the resulting diagnostics json file. It should contain entries for all diagnostics from the build.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
